### PR TITLE
Feat/waheed

### PR DESCRIPTION
**Pull request by waheed94**
userController and committeController are finished

### CODE REVIEW

File: app/controllers/CommitteeController.js [hunk 1]
```diff
@@ -82,30 +86,73 @@ module.exports.show = function(req, res, next){
...
-      }
 
-      next();
+         next();
+
+       }).catch(function(err){
+           /* failed to retrieve the members from the database */
+      		res.status(500).json({
+      			status:'failed',
+      			message: err.message
```
Comment (KEMDAK) [inline app/controllers/CommitteeController.js]: change the message to 'Internal server error'

File: app/controllers/CommitteeController.js [hunk 2]
```diff
@@ -82,30 +86,73 @@ module.exports.show = function(req, res, next){
...
+       });
+
+     }
+    
    }).catch(function(err){
       /* failed to retrieve the committee from the database */
       res.status(500).json({
          status:'failed',
-         message: 'Internal server error'
+         message: err.message
```
Comment (KEMDAK) [inline app/controllers/CommitteeController.js]: change the message to 'Internal server error'

File: app/controllers/CommitteeController.js [hunk 3]
```diff
@@ -146,58 +207,92 @@ module.exports.store = function(req, res, next){
...
-            });
+     var users = req.body.members ;
+     users.push(req.body.head_id);
+
+     User.findAll({where : {id :{in :users}}}).then(function(members){
+          for (var i = members.length - 1; i >= 0; i--) {
+            if(members[i].type == 'Admin' || members[i].type == 'Upper Board' || ( members[i].type == 'High Board' && members[i].id != req.body.head_id)){
+             res.status(400).json({
+               status: 'failed',
+               errors: {param:'members',type:'validity',value:members.id}
```
Comment (KEMDAK) [inline app/controllers/CommitteeController.js]: what is members.id !!!

File: app/controllers/CommitteeController.js [hunk 4]
```diff
@@ -146,58 +207,92 @@ module.exports.store = function(req, res, next){
...
+            			    		message: 'Internal Server Error'
+            			    	});
+
+            			    	req.err = 'CommitteeController.js, Line: 196\nCouldn\'t save the committee in the database.\n' + String(err);
+            			    }
+            			    next();
+            			});
+
+     }).catch(function(err){
+       	res.status(400).json({
```
Comment (KEMDAK) [inline app/controllers/CommitteeController.js]: the status code should be 500

File: app/controllers/CommitteeController.js [hunk 5]
```diff
@@ -244,29 +354,67 @@ module.exports.update = function(req, res, next){
...
 
-      if(affectedRows === 0){
-         /* committee was not found */
-         res.status(400).json({
+   Committee.findById(req.params.id).then(function(committee){
+       if(!committee){
+       	    res.status(404).json({
             status:'failed',
-            message: 'The requested route was not found.'
+            message: 'Internal server error'
```
Comment (KEMDAK) [inline app/controllers/CommitteeController.js]: the message should be 'The requested route was not found.'

File: app/controllers/CommitteeController.js [hunk 6]
```diff
@@ -244,29 +354,67 @@ module.exports.update = function(req, res, next){
...
       }
-      else{
-         /* the committee is updated successfully in the database */
+        
+        var users = req.body.members ; 
+        var head  = req.body.head_id ;
+        if(!req.body.head_id){
+           committee.head(function(user,err){
+                 if(err){
+                  console.log("a7a");
```
Comment (KEMDAK) [inline app/controllers/CommitteeController.js]: e7na fe code mo7taram!!

return an Internal server error response

File: app/controllers/CommitteeController.js [hunk 7]
```diff
@@ -244,29 +354,67 @@ module.exports.update = function(req, res, next){
...
+        }
+
+       
+ function updateCom(){
+      User.findAll({where : {id :{in:users}}}).then(function(members){
+      for (var i = members.length - 1; i >= 0; i--) {
+        if(members[i].type == 'Admin' || members[i].type == 'Upper Board' || ( members[i].type == 'High Board' && members[i].id != head)){
+               res.status(400).json({
+               status: 'failed',
+               errors: {param:'members',type:'validity',value:members.id}
```
Comment (KEMDAK) [inline app/controllers/CommitteeController.js]: same question what is members.id!!

File: app/controllers/CommitteeController.js [hunk 8]
```diff
@@ -290,12 +438,104 @@ module.exports.update = function(req, res, next){
...
          });
 
          req.err = 'CommitteeController.js, Line: 296\nCouldn\'t update the committee in the database.\n' + String(err);
       }
 
       next();
+     });
+      
+    }).catch(function(err){
+        res.status(400).json({
```
Comment (KEMDAK) [inline app/controllers/CommitteeController.js]: the status code should be 500

File: app/controllers/CommitteeController.js [hunk 9]
```diff
@@ -290,12 +438,104 @@ module.exports.update = function(req, res, next){
...
+          error: 'Internal Server Error' 
+        });
+        req.err = 'CommitteeController.js, Line: 143\ncan not find members in the database.\n' + JSON.stringify(errors);
+        next();
+     });
+    };
+
+   }).catch(function(err){
+    
+         res.status(400).json({
```
Comment (KEMDAK) [inline app/controllers/CommitteeController.js]: i think this one should be an internal server error make sure it is correct

File: app/controllers/UserController.js [hunk 10]
```diff
@@ -119,105 +143,126 @@ module.exports.show = function(req, res, next) {
...
 
+         next();
+
       }).catch(function(err) {
-         /* failed to find the user's committee.*/
+         /* failed to find the user's joined tables.*/
          res.status(500).json({
             status:'failed',
-            message: 'Internal server error'
+            message: err.message+"aa"
```
Comment (KEMDAK) [inline app/controllers/UserController.js]: change it to internal server error

File: app/controllers/UserController.js [hunk 11]
```diff
@@ -322,10 +379,72 @@ module.exports.store = function(req, res, next) {
...
+                transporter.sendMail(mailOptions);
+
+
+            });
+
+          
+            var dirPath = path.resolve( '../IEEEGUCSB-System/public/images/'+user.id);
+            fse.ensureDir(dirPath, function(err) { 
+               if(err){
+               res.status(400).json({
```
Comment (KEMDAK) [inline app/controllers/UserController.js]: this is an Internal server error

File: app/routes/UserResource.js [hunk 12]
```diff
@@ -60,17 +65,71 @@ module.exports = function(app) {
     *       gender: the user's gender,       [for detailed view only]
     *       phone_number: the user's phone number,       [for detailed view only]
     *       birthdate: the user's birthdate,        [for detailed view only]
-    *       IEEE_membership_ID: the user's membership id in IEEE,
-    *       settings: the user's profile settings        [public settings only for basic view]
+    *       IEEE_membership_ID: the user's membership id in IEEE,       
+    *       settings: {  the user's profile settings  
+    *             public: {                                         [public settings only for basic view]
```
Comment (KEMDAK) [inline app/routes/UserResource.js]: for all views (basic, detailed and mine)

File: app/controllers/UserController.js [hunk 13]
```diff
@@ -388,7 +507,7 @@ module.exports.update = function(req, res, next) {
    if (req.body.IEEE_membership_ID) {
       req.sanitizeBody('IEEE_membership_ID').escape();
       req.sanitizeBody('IEEE_membership_ID').trim();
-      obj.IEEE_membership_ID = req.body.IEEE_membership_ID;
+      objIEEE_membership_ID = req.body.IEEE_membership_ID;
```
Comment (KEMDAK) [inline app/controllers/UserController.js]: i think this should be `obj.IEEE_membership_ID`

File: app/controllers/CommitteeController.js (hunk 6 above)
Comment (waheed94) [inline app/controllers/CommitteeController.js]: 😂😂 lamo 25za

Comment (waheed94) [inline app/controllers/CommitteeController.js]: 2eda di mesh mwgooda fl code 3ndi 3la fekra ezai tyb !!

File: app/routes/UserResource.js (hunk 12 above)
Comment (waheed94) [inline app/routes/UserResource.js]: mesh fahm ya3ni 23ml 2e

Comment (KEMDAK) [inline app/routes/UserResource.js]: The public setting is available in all views not only the basic

File: app/controllers/UserController.js (hunk 11 above)
Comment (KEMDAK) [inline app/controllers/UserController.js]: @waheed94 Please change this to an internal server error!

**Pull request merged**